File: src/java/VRE.Vridge.API/VRE.Vridge.API.Client/src/main/java/com/riftcat/vridge/api/client/java/control/requests/ControlRequestHeader.java
package com.riftcat.vridge.api.client.java.control.requests;

import com.riftcat.vridge.api.client.java.control.BaseControlMessage;
import com.riftcat.vridge.api.client.java.control.ControlRequestCode;

public class ControlRequestHeader extends BaseControlMessage {
    public String RequestingAppName;

    public ControlRequestHeader(String requestingAppName){
        RequestingAppName = requestingAppName;
    }

    public ControlRequestHeader(int reqCode){
        Code = reqCode;
    }
}

File: src/csharp/VRE.Vridge.API.Client/Messages/Control/Requests/ControlRequestHeader.cs
namespace VRE.Vridge.API.Client.Messages.Control.Requests
{
    public class ControlRequestHeader : BaseControlMessage
    {
        public string RequestingAppName;
        
        public ControlRequestHeader(string requestingAppName, ControlRequestCode code) : base(3, (int)code)
        {
            RequestingAppName = requestingAppName;            
        }

        public ControlRequestHeader() : base()
        {
            // Deserialization only
        }
    }
}
